simtools/corsika/corsika_output.py

```diff
@@ -976,44 +1036,416 @@ def num_photons_per_event(self):
+        return hist.reshape(1, bins), edges.reshape(1, bins + 1)
 
+    def get_num_photons_per_telescope_distr(self, bins=50, hist_range=None):
+        """
+        Get the distribution of photons per telescope.
 
-        Raises
-        ------
-        ValueError:
-            if event_or_telescope not valid.
+        Parameters
+        ----------
+        bins: float
+            Number of bins for the histogram.
+        hist_range: 2-tuple
+            Tuple to define the range of the histogram.
+
+        Returns
+        -------
+        numpy.ndarray
+            The counts of the histogram.
+        numpy.array
+            Number of photons per telescope.
         """
-        if event_or_telescope == "event":
-            hist, edges = np.histogram(self.num_photons_per_event, bins=bins, range=hist_range)
-        elif event_or_telescope == "telescope":
-            hist, edges = np.histogram(self.num_photons_per_telescope, bins=bins, range=hist_range)
-        else:
-            msg = "`event_or_telescope` has to be either 'event' or 'telescope'."
-            self._logger.error(msg)
-            raise ValueError
-        return hist, edges
+
+        hist, edges = np.histogram(self.num_photons_per_telescope, bins=bins, range=hist_range)
+        return hist.reshape(1, bins), edges.reshape(1, bins + 1)
+
+    def export_histograms(self, output_dir="./"):
+        """
+        Export the histograms to ecsv files.
+
+        Parameters
+        ----------
+        output_dir: str
+            Output directory, where to save the histograms.
+        """
+
+        self._export_1D_histograms(output_dir=output_dir)
+        self._export_2D_histograms(output_dir=output_dir)
+
+    @property
+    def _meta_dict(self):
+        """
+        Define the meta dictionary for exporting the histograms.
+
+        Returns
+        -------
+        dict
+            Meta dictionary for the ECSV files with the histograms.
+        """
+
+        if self.__meta_dict is None:
+            self.__meta_dict = {
+                "CORSIKA version": self.version,
+                "simtools version": version.__version__,
+                "Original IACT file": self.input_file.name,
+                "telescope_indices": list(self.telescope_indices),
+                "individual_telescopes": self.individual_telescopes,
+                "msg": "Only lower bin edges are given.",
+            }
+        return self.__meta_dict
+
+    @property
+    def _dict_1D_distributions(self):
+        """
+        Dictionary to label the 1D distributions according to the class methods.
+
+        Returns
+        -------
+        dict:
+            The dictionary with information about the 1D distributions.
+        """
+        self.__dict_1D_distributions = {
+            "wavelength": {
+                "function": "get_photon_wavelength_distr",
+                "file name": "hist_1D_photon_wavelength_distr",
+                "title": "Wavelength distribution",
```

Comment:
I think the title should include the word "photon" everywhere. E.g., "Photon wavelength distribution", "Radial photon distribution on the ground", etc.

Reply:
Ok, I am adding it.